tests/Aspire.Dashboard.Tests/Utils/CertificateUtilTests.cs

```diff
@@ -0,0 +1,132 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNetCore.InternalTesting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Xunit;
+
+namespace Aspire.Dashboard;
+
+public sealed class CertificateUtilTests
+{
+    [Theory]
+    [InlineData("testCert.pfx", X509ContentType.Pkcs12)]
+    [InlineData("https-dsa.pem", X509ContentType.Cert)]
+    public void TestCertificateTypes(string fileName, X509ContentType contentType)
+    {
+        var filePath = TestCertificateLoader.GetCertPath(fileName);
+
+        Assert.Equal(contentType, X509Certificate2.GetCertContentType(filePath));
+    }
+
+    [Fact]
+    public void GetFileCertificate_Pkcs12_CorrectPassword()
+    {
+        var certificates = CertificateUtil.GetFileCertificate(
+            filePath: TestCertificateLoader.GetCertPath("testCert.pfx"),
+            password: "testPassword",
+            logger: NullLogger.Instance);
+
+        Assert.NotNull(certificates);
+
+        var cert = Assert.Single(certificates.Cast<X509Certificate>());
+
+        Assert.NotNull(cert);
+        Assert.Equal("CN=localhost", cert.Subject);
+    }
+
+    [Fact]
+    public void GetFileCertificate_Pkcs12_IncorrectPassword()
+    {
+        Assert.Throws<CryptographicException>(() => CertificateUtil.GetFileCertificate(
+            filePath: TestCertificateLoader.GetCertPath("testCert.pfx"),
+            password: "wrongPassword",
+            logger: NullLogger.Instance));
+    }
+
+    [Fact]
+    public void GetFileCertificate_Cert_WithRedundantPassword()
+    {
+        var filePath = TestCertificateLoader.GetCertPath("https-dsa.pem");
```

Comment:
It's a little weird to me that you're testing with DSA, since that algorithm itself is now entirely obsolete.  (FIPS 186-5 withdrew it, so it's not just "the least popular", it's "RIP")